src/main/java/org/reaktivity/nukleus/mqtt/internal/stream/MqttServerFactory.java

```diff
@@ -660,21 +692,15 @@ else if (!"MQTT".equals(mqttConnect.protocolName().asString()) || mqttConnect.pr
                 {
                     reasonCode = UNSUPPORTED_PROTOCOL_VERSION;
                 }
-                else if (isSetWillFlag(flags))
-                {
-                    reasonCode = MALFORMED_PACKET;
-                }
                 else if (isSetBasicAuthentication(flags))
                 {
                     reasonCode = NOT_AUTHORIZED;
                 }
```

Comment:
This section can be hoisted to a separate method and called via `reasonCode = decodeConnectFlags(flags);`